src/Context/product-context.js

```diff
@@ -13,7 +13,7 @@ const useProducts = () => useContext(ProductContext);
 const ProductProvider = ({ children }) => {
   const [state, dispatch] = useReducer(reducer, [{}]);
```

Comment:
It would be better to name your ```state``` as ```filterState``` as will have multiple reducers. So it will be easier to distinguish them.

Reply:
Done, Thank you Kevin